src/reduced_basis/rb_data_deserialization.C

```diff
@@ -90,10 +92,14 @@ void RBEvaluationDeserialization::read_from_file(const std::string & path,
   capnp::ReaderOptions reader_options;
   reader_options.traversalLimitInWords = std::numeric_limits<uint64_t>::max();
 
-  std::unique_ptr<capnp::StreamFdMessageReader> message;
+  std::unique_ptr<capnp::InputStreamMessageReader> message;
   libmesh_try
     {
-      message = std::make_unique<capnp::StreamFdMessageReader>(fd, reader_options);
+      // Define the reader type, based on whether or not packing is used.
+      if(use_packing)
+        message = std::make_unique<capnp::PackedFdMessageReader>(fd, reader_options);
+      else
+        message = std::make_unique<capnp::StreamFdMessageReader>(fd, reader_options);
```

Comment:
It occurred to me as well, but since the two `make_uniques` return different types, it didn't compile.

There's probably some way to cast the return in the ternary so it works, but I didn't try any further.

Reply:
Oh, yes of course the types are different but they are caught as a `unique_ptr` to the base type.

Reply:
Yeah. I don't think using a ternary operator and casting is worth it here, it's just going to degrade readability.